### PR TITLE
Feature/vehicle e2e tests post

### PR DESCRIPTION
**Pull request by wangscs**

### CODE REVIEW

File: VehicleWebAPI/Services/VehicleDatabaseService.cs [hunk 1]
```diff
@@ -5,50 +5,67 @@
...
+      if (vehicle.VIN == null
+      || vehicle.Make == null
+      || vehicle.Model == null
+      || vehicle.Year == null
+      || vehicle.Miles < 0
+      || vehicle.Color == null
+      || vehicle.SellingPrice < 0
+      || vehicle.Status < 0
+      || vehicle.UserId < 0) return 0;
+      if (_db.Vehicles.FirstOrDefault(t => t.VIN == vehicle.VIN) != null) return 0;
```
Comment (rmuehler) [inline VehicleWebAPI/Services/VehicleDatabaseService.cs:29]: Think the logic needs to be a bit different here, things like Year will default to 0 and not be null. So maybe year > 1900? Also need to think about other validation.

**Review by rmuehler — APPROVED** [review]
:)

File: VehicleWebAPI/Services/VehicleDatabaseService.cs (hunk 1 above)
Comment (wangscs) [reply VehicleWebAPI/Services/VehicleDatabaseService.cs:29]: Will be addressed in a future PR :)

**Review by rmuehler — APPROVED** [review]

**Review by GannonDeWalt — APPROVED** [review]
looks good

**Review by ammitchky — APPROVED** [review]
looks good

**Pull request merged**